lib/elixir/lib/map.ex

```diff
@@ -971,4 +971,115 @@ defmodule Map do
   def size(map) do
     map_size(map)
   end
+
+  @doc """
+  Filters `map`, i.e. returns a map containing
+  only those elements for which the function `fun` returns a truthy value.
+
+  `fun` receives the key and value of each of the
+  elements in the map as a `{key, value}` tuple.
+
+  `Map.filter/2` is faster than using `map |> Enum.filter(fun) |> Enum.into(%{})`,
+  as no intermediate list is being built.
+
+
+  See also `reject/2` which discards all elements where the
+  function returns a truthy value.
+
+  ## Examples
+
+      iex> Map.filter(%{one: 1, two: 2, three: 3}, fn {_key, val} -> rem(val, 2) == 1 end)
+      %{one: 1, three: 3}
+
+  """
+  @doc since: "1.13.0"
+  @spec filter(map, ({key, value} -> as_boolean(term))) :: map
+  def filter(map, fun) when is_map(map) and is_function(fun, 1) do
+    iter = iterator(map)
+    next = next(iter)
+    :maps.from_list(do_filter(next, fun))
+  end
+
+  defp do_filter(:none, _fun), do: []
+
+  defp do_filter({key, value, iter}, fun) do
+    if fun.({key, value}) do
+      [{key, value} | do_filter(next(iter), fun)]
+    else
+      do_filter(next(iter), fun)
+    end
+  end
+
+  @doc """
+  Returns map excluding the elements from `map` for which
+  the function `fun` returns a truthy value.
+
+
+  `Map.reject/2` is faster than using `map |> Enum.reject(fun) |> Enum.into(%{})`,
+  as no intermediate list is being built.
+
+  See also `filter/2` which discards all elements where the
+  function returns a falsy value.
```

Comment:
```suggestion
  See also `filter/2`.
```

Reply:
I did not make this change, because Enum.reject/2 also contains this longer sentence.
Should we keep both long sentences, or also alter Enum.reject/2's docs?

Reply:
`Enum.reject/2` is short, `Enum.filter` is long: https://github.com/elixir-lang/elixir/blob/d6d8ad419f8fe91dda7eb3f74e34041a924bd7f9/lib/elixir/lib/enum.ex#L2444 this is why I suggested

Reply:
Ah, I see :smile: ! :+1: